byzcoin/service.go

```diff
@@ -2348,7 +2370,7 @@ var existingDB = regexp.MustCompile(`^ByzCoin_[0-9a-f]+$`)
 func newService(c *onet.Context) (onet.Service, error) {
 	s := &Service{
 		ServiceProcessor:       onet.NewServiceProcessor(c),
-		contracts:              make(map[string]ContractFn),
+		contracts:              globalContractRegistry.Clone(),
```

Comment:
```suggestion
		contracts:              GetContractRegistry(),
```

Reply:
The public returns a read-only registry but the service should have the writable type.